tsdb/wal/wal.go

```diff
@@ -616,7 +616,10 @@ func (w *WAL) log(rec []byte, final bool) error {
 
 	// Compress the record before calculating if a new segment is needed.
 	compressed := false
-	if w.compress && len(rec) > 0 {
+	if w.compress &&
+		len(rec) > 0 &&
+		// If MaxEncodedLen is less than 0 the record is too large to be compressed.
+		snappy.MaxEncodedLen(len(rec)) >= 0 {
```

Comment:
I am wondering if we could optimize that in the future and only compress if MaxEncodedLen < len(rec). I have no idea how often we would avoid the snappy encryption.

Reply:
but that is out of scope for this PR

Reply:
MaxEncodedLen is always greater than len(rec) since it is possible for snappy to produce results larger than the input (say if the input was completely random). A bit below this we only use the compressed result if the compressed record is actually smaller than the raw record.